pkg/controller/gitrepo/gitrepo_reconcile.go

```diff
@@ -43,7 +43,12 @@ func (r *ReconcileGitRepo) Reconcile(request reconcile.Request) (reconcile.Resul
 			}
 			return err
 		}
-		helpers.AddTenantLabel(&instance.ObjectMeta, instance.Spec.TenantRef.Name)
+		if helpers.AddTenantLabel(&instance.ObjectMeta, instance.Spec.TenantRef.Name) {
+			err := r.client.Update(context.TODO(), instance)
```

Comment:
I fear we have to call `Update` further down since we add/change other things later which won't ever be written like that...

Reply:
Right, moved it to the bottom of the retry block.